source/postprocess/point_values.cc

```diff
@@ -0,0 +1,270 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/point_values.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <math.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    PointValues<dim>::execute (TableHandler &)
+    {
+      // evaluate the solution at all of our evaluation points
+      std::vector<Vector<double> >
+      current_point_values (evaluation_points.size(),
+                            Vector<double> (this->introspection().n_components));
+
+      for (unsigned int p=0; p<evaluation_points.size(); ++p)
+        {
+          // try to evaluate the solution at this point. in parallel, the point
+          // will be on only one processor's owned cells, so the others are
+          // going to throw an exception. make sure at least one processor
+          // finds the given point
+          bool point_found = false;
+          try
+            {
+              VectorTools::point_value(this->get_dof_handler(),
+                                       this->get_solution(),
+                                       evaluation_points[p],
+                                       current_point_values[p]);
+              point_found = true;
+            }
+          catch (const VectorTools::ExcPointNotAvailableHere &)
+            {
+              // ignore
+            }
+
+          // ensure that exactly one processor found things
+          Assert (Utilities::MPI::sum (point_found ? 1 : 0, this->get_mpi_communicator()) > 0,
```

Comment:
That's a good question. I opted on the safe side, but you're right -- since I add up contributions below, I need to check that exactly one processor found that point, or we get into trouble.